src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIService.java

```diff
@@ -53,6 +66,29 @@ public void createBuildPlanForExercise(ProgrammingExercise programmingExercise,
         // which results in a new build being triggered.
         // For local CI, a build plan must not be created, because all the information for building a submission and running tests is contained in the participation, so we only
         // trigger the build here.
+
+        // Check if docker image exists locally, otherwise pull it from Docker Hub.
+
+        try {
+            log.info("Inspecting docker image {}", dockerImage);
+            dockerClient.inspectImageCmd(dockerImage).exec();
+        }
+        catch (NotFoundException e) {
+            // Image does not exist locally, pull it from Docker Hub.
+            log.info("Pulling docker image {}", dockerImage);
+            try {
+                dockerClient.pullImageCmd(dockerImage).exec(new PullImageResultCallback()).awaitCompletion();
+            }
+            catch (InterruptedException ie) {
+                throw new LocalCIException("Interrupted while pulling docker image " + dockerImage, ie);
+            }
+        }
+        catch (BadRequestException e) {
+            throw new LocalCIException("Error while inspecting docker image " + dockerImage, e);
+        }
+
+        // Trigger build for the given participation.
```

Comment:
Can you reuse the code from `LocalCIBuildJobManagementService` here (or vice versa)? This is a bit duplicated here.

Reply:
Moved the method to a new class.